RS/JX/Rule5-43-2022

Jiarongx-Xie opened the pull request:
This rule has not been tested.

Jiarongx-Xie committed:
initial setup

Jiarongx-Xie committed:
initial dev

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012022/section5/section5rule43.py:
```diff
@@ -0,0 +1,65 @@
+
+
+class PRM9012019Rule86r63(RuleDefinitionListIndexedBase):
+    """Rule 11 of ASHRAE 90.1-2019 Appendix G Section 6 (Lighting)"""
```
Should we update lines 11 and 12?

Jiarongx-Xie replied:
Good point!

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012022/section5/section5rule43.py:
```diff
@@ -0,0 +1,65 @@
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            automatic_shades_modeled = calc_vals["automatic_shades_modeled"]
+            return len(automatic_shades_modeled) > 0
```
In the RDS (I was looking into [this file](https://github.com/pnnl/ruleset-checking-tool/blob/c4687b9d1fdfdae4e12d893ea19c5d3436f332c7/docs/section5/Rule5-43.md)), it says ```if len(automatic_shades_modeled) == 0``` is a pass. Could you double-check on this?

Jiarongx-Xie replied:
Yes! I have corrected this.

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012022/section5/section5rule43.py:
```diff
@@ -0,0 +1,65 @@
+            super(PRM9012019Rule86r63.BuildingRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=False, BASELINE_0=True, PROPOSED=False
+                ),
```
Fail message is missing - Should we add a fail message here?

Jiarongx-Xie replied:
Good catch, added!

yunjoonjung-PNNL commented on rct229/rulesets/ashrae9012022/section5/section5rule43.py:
```diff
@@ -0,0 +1,65 @@
+        def get_calc_vals(self, context, data=None):
+            building_b = context.BASELINE_0
+            automatic_shades_modeled = [
+                subsurface_b.get("has_automatic_shades")
```
Since we already know ```has_automatic_shades``` value exists from line 59, should we use the ```subsurface_b["has_automatic_shades"]``` code?

Jiarongx-Xie replied:
Good point, updated!

yunjoonjung-PNNL reviewed (approved):
I've reviewed only 5-43 rule file and left a few comments. Please double-check these. Thank you Jiarong!

Jiarongx-Xie commented:
> I've reviewed only 5-43 rule file and left a few comments. Please double-check these. Thank you Jiarong!

I have addressed all your comments. Thank you Yun Joon!

Jiarongx-Xie committed:
address PR comments

jugonzal07 committed:
First cut at 5-43 in 90.1-2022

jugonzal07 committed:
Removed unnecessary HVAC systems.

yunjoonjung-PNNL committed:
Corrected rule logic

jugonzal07 committed:
Correcting pass/fail cases in 5-43. I got them flipped.

yunjoonjung-PNNL committed:
Merge branch 'RCT/YJ/rule_testing_2022' of https://github.com/pnnl/ruleset-checking-tool into RS/JX/Rule5-43-2022

yunjoonjung-PNNL committed:
Merge pull request #1790 from pnnl/RT/JG/ruletest_5_43

Rule Test 5-43 90.1-2022

yunjoonjung-PNNL committed:
Merge branch 'feature/ashrae-9012022' of https://github.com/pnnl/ruleset-checking-tool into RS/JX/Rule5-43-2022

weilixu committed:
Merge branch 'feature/ashrae-9012022' into RS/JX/Rule5-43-2022

weilixu commented on rct229/rulesets/ashrae9012022/section5/section5rule43.py:
```diff
@@ -0,0 +1,92 @@
+                )
+                if surface_b.get("adjacent_to") == SURFACE_ADJACENCY.EXTERIOR
+                for subsurface_b in surface_b.get("subsurfaces", [])
+                if subsurface_b.get("has_automatic_shades")
```
I think we do not need this if,
in the object:
```python
{
      "has_automatic_shades": subsurface_b.get("has_automatic_shades", False),
      "id": subsurface_b["id"],
}
```

yunjoonjung-PNNL replied:
Addressed.

weilixu reviewed (approved):
Approved with one comment needs to be addressed.

yunjoonjung-PNNL committed:
Addressed PR comment

yunjoonjung-PNNL committed:
Removed .get line
